.github/src/EngineCompute.java

```diff
@@ -34,39 +28,71 @@ public String compute(int num) {
                 num /= 16;
             }
         }
-        
-        return "0x" + hexNum.toString();
+
+        return "0x" + hexNum.toString(); //maybe remove toString()?
+
+//        if (num == 0) {
+//            return "0x0";
+//        }
+//
+//        char[] hexChars = new char[8]; // Assuming integers are 32-bit
+//        int i = 0;
+//
+//        while (num != 0 && i < 8) {
+//            int remainder = num % 16;
+//            hexChars[i++] = getHexDigit(remainder);
+//            num /= 16;
+//        }
+//
+//        reverseArray(hexChars, i);
+//
+//        return "0x" + new String(hexChars, 0, i);
     }
 
-    private char getHexDigit(int remainder) {
+        private void reverseArray(char[] hexChars, int i) {
+
+            int start = 0;
+            int end = i - 1;
+            while (start < end) {
+                char temp = hexChars[start];
+                hexChars[start] = hexChars[end];
+                hexChars[end] = temp;
+                start++;
+                end--;
+            }
+        }
+
+        private char getHexDigit(int remainder) {
         
         char digit = '0';
 
         if(remainder < 10){
 
             digit = (char) ('0' + remainder); //takes the character 0(ASCII) and adds the remainder to get the remainder ASCII value
-        }else if(remainder >= 10){ //can only be betwen 10 & 15 since were dividing by 16
+
+        }else if(remainder >= 10){ //can only be between 10 & 15 since were dividing by 16
 
             digit = (char) ('A' + remainder - 10); //finds correct ASCII value
         }
 
         return digit;
     }
 
-    //finds greatest common divisor of a list of integers
-    public int getGCD(List<Integer> decInputs) {
-
-        int gcd = decInputs.get(0);
-
-        for(int i = 1; i < decInputs.size(); i++) {
-            gcd = getGCD(gcd, decInputs.get(i));
-        }
-
-        return gcd;
-    }
-
-    //finds GCD of two numbers using Euclid's algorithmn
-    private int getGCD(int numOne, int numTwo) {
+    /***** COMMENTED OUT / NOT USED TO IMPROVE CPU USAGE *****/
```

Comment:
Makes sense with the change in Engine Manager to reduce the amount of data reading and interations.